### ISSUE TITLE
[5.6.0] Memory leak when moving STL model with arrows

### ISSUE THREAD

**Issue by sharky98** [Ultimaker/Cura#18716]
### Cura Version

5.6.0

### Operating System

Windows 11

### Printer

Not applicable (but if even, Ender 3 S1 Pro)

### Reproduction steps

1. Fresh install of Cura.
2. Disable **all** bundled plugins.
3. Restart Cura.
4. Open STL file (such as `Benchy.stl`).
5. Move STL file with arrows.

### Actual results

For the meme 😉

<img src="https://github.com/Ultimaker/Cura/assets/8150191/d54bba5d-a1d2-4b76-9435-bc3d2ef464c7" height="100" /><br /><br />

But with seriousness, on start, Cura has a working set of *~400MB*. After loading `Benchy.stl` (which is ~11MB), Cura is up to *~540MB* of working set (varies approx. ±10MB depending on the camera position and if the model is selected or not). But then, if I take any axis to move by hand (let's say X from 0 to -66 and whatever decimal my mouse will end up on), the working set shoots up to *~810MB*.

So, that means there is around 270MB of data in memory to deal with the X translation of approx. 66mm 🤯! Move it around a few more and you get eat up all the memory in no time!

Note: Undoing the movement, Unselecting the model, or deleting the model does not free up the memory; in fact, if anything it increase (to cover the "undo stack" I suspect).

### Expected results

Moving a model does not eat up all the memory (or free it once the move is done).

### Add your .zip and screenshots here ⬇️

I made a screen capture and provide two log files to help out debugging.

I did try to check with Process Explorer and Process Monitor if there was anything suspect to an untrained eye (such as quantity of handles, threads, objects, etc. that increase alongside the memory usage), but came out empty-handed.

## Screen Capture

This is a bit more extensive than the single Benchy above, but also shows that it seems that rotation is not affected by this memory leak. Moreover, I did have 3rd party plugins, etc.

[cura.log](https://github.com/Ultimaker/Cura/files/14738514/cura-video.log)

https://github.com/Ultimaker/Cura/assets/8150191/92ff996f-fd6c-402d-a54a-696f62521353

## Single Benchy Test

This one is without any plugins (including all supplied plugins disabled). No fancy screen capture for this one though. Not much going on in the logs to my eyes, expect that there is a lot of `DEBUG - [MainThread] UM.Operations.OperationStack.push [72]: TranslateOp.(node=<CuraSceneNode object: '3DBenchy.stl'>,trans.=<0.000,0.000,-0.XXX>), took Xms`.

Could the logging be the culprit? Even though I don't think the 20-ish lines of debug logs could account for the ~210MB of memory used!

[cura.log](https://github.com/Ultimaker/Cura/files/14738578/cura-after-closing.log)

![image](https://github.com/Ultimaker/Cura/assets/8150191/d885b2b4-470f-414f-9c61-2416d9b79aa6)

![image](https://github.com/Ultimaker/Cura/assets/8150191/c3c0b03a-16ba-4bc2-92e3-383727d1383e)

![image](https://github.com/Ultimaker/Cura/assets/8150191/8ac8c836-44d0-4c59-9599-a74444c96a06)

![image](https://github.com/Ultimaker/Cura/assets/8150191/69a2e4c2-b2e6-4127-8b01-cf03ae557b29)

**Comment by sharky98** [comment]
This may be a duplicate of #18446, but since they didn't mention anything about moving with the arrows, I opted to created a new one.

**Comment by sharky98** [comment]
For completeness, I tested the following old version (I am new, so I cannot tell when it last worked).

For them, I deleted my configuration folder and started from scratch using the default custom FFF printer. And I re-did 5.6.0 "just-in-case".

* ✅ 4.7.0
* ✅ 4.13.1
* ✅ 5.0.0
* ✅ 5.4.0 (damn... I starting to second-guess my initial thinking that my printer settings had nothing to do with the bug! Let's way to the redo of 5.6.0...)
* ❌ 5.6.0 (oof! not working... so the printer profile has nothing to do with this!)
* ❌ 5.7.0-beta.1
